authors/apps/articles/views.py

```diff
@@ -142,3 +143,122 @@ def put(self, request, slug, *args, **kwargs):
                     serializer.errors,
                     status=400
                 )
+
+
+def get_article(slug):
+    """
+        Returns specific article using slug
+    """
+    article = Article.objects.all().filter(slug=slug).first()
+    if article is None:
+        raise exceptions.NotFound({
+            "message": error_msgs['not_found']
+        }, status.HTTP_404_NOT_FOUND)
+    return article
+
+
+class ShareViaEmail(generics.CreateAPIView):
+    """
+        Share Articles via email
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, slug):
+        """
+            POST /api/v1/<slug>/share/email/
+        """
+        email = request.data['email']
+        get_article(slug)
+
+        if not email:
+            return Response({
+                'message': error_msgs['no_email'],
+            },
+                status=status.HTTP_400_BAD_REQUEST)
+        elif re.search(r"^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$", email) is None:
+            return Response({
+                'message': error_msgs['email_format']
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        username = request.user
+
+        # format the email
+        host = request.get_host()
+        protocol = request.scheme
+        shared_link = protocol + '://' + host + '/api/v1/articles/' + slug
+        subject = "Authors Haven"
+        message = render_to_string(
```

Comment:
@BoltC0rt3z nice work however it would make more to add the title of the article as well in the share message.